// zmp-ui/esm/components/bottom-navigation/index.stories.js
import React from "react";
import BottomNavigation from ".";
import Icon from "../icon"; // Assuming Icon is available in the project

var meta = {
  title: "Layout/BottomNavigation",
  component: BottomNavigation,
  parameters: {
    layout: "centered",
    docs: {
      story: {
        inline: false
      }
    }
  },
  tags: ["autodocs"],
  argTypes: {}
};
export default meta;
export var Default = {
  name: "Mặc định",
  args: {
    fixed: true,
    children: [/*#__PURE__*/React.createElement(BottomNavigation.Item, {
      key: "/",
      label: "Tin nh\u1EAFn",
      icon: /*#__PURE__*/React.createElement(Icon, {
        icon: "zi-chat"
      })
    }), /*#__PURE__*/React.createElement(BottomNavigation.Item, {
      key: "/notification",
      label: "Th\xF4ng b\xE1o",
      icon: /*#__PURE__*/React.createElement(Icon, {
        icon: "zi-notif"
      })
    }), /*#__PURE__*/React.createElement(BottomNavigation.Item, {
      key: "/profile",
      label: "C\xE1 nh\xE2n",
      icon: /*#__PURE__*/React.createElement(Icon, {
        icon: "zi-user"
      })
    }), /*#__PURE__*/React.createElement(BottomNavigation.Item, {
      key: "/settings",
      label: "C\xE0i \u0111\u1EB7t",
      icon: /*#__PURE__*/React.createElement(Icon, {
        icon: "zi-setting"
      })
    })]
  }
};
