// zmp-ui/cjs/components/modal/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.InfoDialogVerticalActions = exports.InfoDialog = exports.CustomModalWithCover = exports.CustomContent = void 0;
var _ = require("../..");
var _react = _interopRequireDefault(require("react"));
var _2 = _interopRequireDefault(require("."));
var meta = {
  title: "Overlay/Modal",
  component: _2["default"],
  tags: ["autodocs"],
  argTypes: {}
};
var _default = exports["default"] = meta;
var InfoDialog = exports.InfoDialog = {
  name: "Mặc định",
  args: {
    visible: true,
    title: "Bạn có muốn đặt bàn tại ZaUI Coffee không?",
    actions: [{
      text: "Xem menu"
    }, {
      text: "Đặt bàn ngay",
      close: true,
      highLight: true
    }],
    description: "Hãy xem qua menu trước khi quyết định đặt bàn. Chúng tôi có nhiều món hấp dẫn đang chờ bạn!"
  }
};
var InfoDialogVerticalActions = exports.InfoDialogVerticalActions = {
  name: "Hiển thị các action theo chiều dọc",
  args: {
    visible: true,
    title: "Bạn có muốn đặt bàn tại ZaUI Coffee không?",
    verticalActions: true,
    actions: [{
      text: "Xem menu chi tiết"
    }, {
      text: "Tiếp tục đặt bàn",
      close: true,
      highLight: true
    }],
    description: "Bạn có thể tham khảo thêm thông tin món ăn trước khi đặt bàn để có trải nghiệm tốt nhất."
  }
};
var CustomModalWithCover = exports.CustomModalWithCover = {
  name: "Hình ảnh cover",
  args: {
    visible: true,
    title: "Ưu đãi đặc biệt từ ZaUI Coffee",
    coverSrc: "https://images.unsplash.com/photo-1561336313-0bd5e0b27ec8",
    zIndex: 1200,
    actions: [{
      text: "Không quan tâm"
    }, {
      text: "Tìm hiểu thêm",
      close: true,
      highLight: true
    }],
    description: "Khám phá các chương trình khuyến mãi hấp dẫn dành riêng cho thành viên khi đặt bàn trực tuyến."
  }
};
var CustomContent = exports.CustomContent = {
  name: "Nội dung tùy chỉnh",
  args: {
    visible: true,
    actions: [{
      text: "Đóng",
      close: true,
      highLight: true
    }],
    coverSrc: "https://images.unsplash.com/photo-1561336313-0bd5e0b27ec8",
    children: /*#__PURE__*/_react["default"].createElement(_.Box, {
      flex: true,
      flexDirection: "column",
      alignItems: "center",
      textAlign: "center"
    }, /*#__PURE__*/_react["default"].createElement(_.Avatar, {
      style: {
        position: "absolute",
        marginTop: -60
      }
    }, "Z"), /*#__PURE__*/_react["default"].createElement(_.Text.Title, null, "ZaUI Coffee"), /*#__PURE__*/_react["default"].createElement(_.Text, {
      size: "small"
    }, "Chu\u1ED7i cafe \u0111\u01B0\u1EE3c y\xEAu th\xEDch s\u1ED1 1 t\u1EA1i Vi\u1EC7t Nam"))
  }
};
