// zmp-ui/cjs/components/button/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithIcon = exports.TertiaryNeutral = exports.TertiaryDanger = exports.Tertiary = exports.SuffixIcon = exports.Submit = exports.Small = exports.SecondaryNeutral = exports.SecondaryDanger = exports.Secondary = exports.Reset = exports.Primary = exports.PrefixIcon = exports.PrefixAndSuffixIcon = exports.Loading = exports.Large = exports.FullWidth = exports.Disabled = exports.Danger = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _icon = _interopRequireDefault(require("../icon"));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; } // ZaUI Icon
// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
var meta = _objectSpread({
  title: "Form/Button",
  component: _["default"],
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {}
}, _parameters.inlineCenterParameters);
var _default = exports["default"] = meta;
// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
var Base = {
  name: "Mặc định",
  args: {
    children: "Thanh toán"
  }
};
var Primary = exports.Primary = {
  name: "Nút chính",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    variant: "primary"
  })
};
var Secondary = exports.Secondary = {
  name: "Nút phụ",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    variant: "secondary"
  })
};
var Tertiary = exports.Tertiary = {
  name: "Nút cấp ba",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    variant: "tertiary"
  })
};
var Danger = exports.Danger = {
  name: "Nút nguy hiểm",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    type: "danger"
  })
};
var SecondaryDanger = exports.SecondaryDanger = {
  name: "Nút phụ nguy hiểm",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    variant: "secondary",
    type: "danger"
  })
};
var TertiaryDanger = exports.TertiaryDanger = {
  name: "Nút cấp ba nguy hiểm",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    variant: "tertiary",
    type: "danger"
  })
};
var SecondaryNeutral = exports.SecondaryNeutral = {
  name: "Nút phụ trung tính",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    variant: "secondary",
    type: "neutral"
  })
};
var TertiaryNeutral = exports.TertiaryNeutral = {
  name: "Nút cấp ba trung tính",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    variant: "tertiary",
    type: "neutral"
  })
};
var Small = exports.Small = {
  name: "Nút nhỏ",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    size: "small"
  })
};
var Large = exports.Large = {
  name: "Nút lớn",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    size: "large"
  })
};
var Loading = exports.Loading = {
  name: "Nút đang tải",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    loading: true
  })
};
var Disabled = exports.Disabled = {
  name: "Nút bị vô hiệu hóa",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    disabled: true
  })
};
var FullWidth = exports.FullWidth = {
  name: "Nút toàn chiều rộng",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    fullWidth: true
  }),
  parameters: {
    layout: "padded"
  }
};
var WithIcon = exports.WithIcon = {
  name: "Nút biểu tượng",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    icon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-star"
    }) // Use ZaUI icon
  })
};
var PrefixIcon = exports.PrefixIcon = {
  name: "Nút có biểu tượng ở đầu",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    prefixIcon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-arrow-left"
    }) // Use ZaUI icon
  })
};
var SuffixIcon = exports.SuffixIcon = {
  name: "Nút có biểu tượng ở cuối",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    suffixIcon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-arrow-right"
    }) // Use ZaUI icon
  })
};
var PrefixAndSuffixIcon = exports.PrefixAndSuffixIcon = {
  name: "Nút có biểu tượng ở đầu và cuối",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    prefixIcon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-arrow-left"
    }),
    suffixIcon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-arrow-right"
    })
  })
};
var Submit = exports.Submit = {
  name: "Nút submit form",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    htmlType: "submit"
  })
};
var Reset = exports.Reset = {
  name: "Nút reset form",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    htmlType: "reset"
  })
};
