// zmp-ui/esm/components/center/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from "react";
import Center from ".";
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Container/Center",
  component: Center,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
var Base = {
  args: {
    style: {
      width: 240,
      border: "1px solid black"
    },
    children: ["#dc1f18", "#34b764", "#fdd949"].map(function (color) {
      return /*#__PURE__*/React.createElement("div", {
        key: color,
        style: {
          backgroundColor: color,
          width: 40,
          height: 40
        }
      });
    })
  }
};
export var Default = {
  name: "Mặc định",
  args: _objectSpread({}, Base.args)
};
export var WithGutters = {
  name: "Gutters",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    gutters: "2rem"
  })
};
export var Intrinsic = {
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    intrinsic: true
  })
};
export var AndText = {
  name: "Và Text",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    andText: true,
    children: "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Modi, unde vitae corporis nesciunt harum, natus asperiores eveniet, perferendis enim dolores porro cupiditate recusandae tenetur doloribus corrupti nobis. Quo, porro perspiciatis."
  })
};
