// zmp-ui/cjs/components/center/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithGutters = exports.Intrinsic = exports.Default = exports.AndText = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Container/Center",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Base = {
  args: {
    style: {
      width: 240,
      border: "1px solid black"
    },
    children: ["#dc1f18", "#34b764", "#fdd949"].map(function (color) {
      return /*#__PURE__*/_react["default"].createElement("div", {
        key: color,
        style: {
          backgroundColor: color,
          width: 40,
          height: 40
        }
      });
    })
  }
};
var Default = exports.Default = {
  name: "Mặc định",
  args: _objectSpread({}, Base.args)
};
var WithGutters = exports.WithGutters = {
  name: "Gutters",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    gutters: "2rem"
  })
};
var Intrinsic = exports.Intrinsic = {
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    intrinsic: true
  })
};
var AndText = exports.AndText = {
  name: "Và Text",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    andText: true,
    children: "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Modi, unde vitae corporis nesciunt harum, natus asperiores eveniet, perferendis enim dolores porro cupiditate recusandae tenetur doloribus corrupti nobis. Quo, porro perspiciatis."
  })
};
