// zmp-ui/cjs/components/bottom-navigation/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _icon = _interopRequireDefault(require("../icon"));
// Assuming Icon is available in the project

var meta = {
  title: "Layout/BottomNavigation",
  component: _["default"],
  parameters: {
    layout: "centered",
    docs: {
      story: {
        inline: false
      }
    }
  },
  tags: ["autodocs"],
  argTypes: {}
};
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    fixed: true,
    children: [/*#__PURE__*/_react["default"].createElement(_["default"].Item, {
      key: "/",
      label: "Tin nh\u1EAFn",
      icon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
        icon: "zi-chat"
      })
    }), /*#__PURE__*/_react["default"].createElement(_["default"].Item, {
      key: "/notification",
      label: "Th\xF4ng b\xE1o",
      icon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
        icon: "zi-notif"
      })
    }), /*#__PURE__*/_react["default"].createElement(_["default"].Item, {
      key: "/profile",
      label: "C\xE1 nh\xE2n",
      icon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
        icon: "zi-user"
      })
    }), /*#__PURE__*/_react["default"].createElement(_["default"].Item, {
      key: "/settings",
      label: "C\xE0i \u0111\u1EB7t",
      icon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
        icon: "zi-setting"
      })
    })]
  }
};
