// zmp-ui/esm/components/progress/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import Progress from ".";
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Display/Progress",
  component: Progress,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
export var Default = {
  name: "Mặc định",
  args: {
    completed: 50,
    maxCompleted: 100
  }
};
export var CustomColors = {
  name: "Màu tùy chỉnh",
  args: {
    completed: 30,
    maxCompleted: 100,
    strokeColor: "green",
    trailColor: "lightgray"
  }
};
export var CustomStrokeWidth = {
  name: "Độ dày tùy chỉnh",
  args: {
    completed: 60,
    maxCompleted: 100,
    strokeWidth: 10
  }
};
export var WithLabel = {
  name: "Có nhãn",
  args: {
    completed: 75,
    maxCompleted: 100,
    showLabel: true
  }
};
