// zmp-ui/esm/components/slider/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import Slider from ".";
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Form/Slider",
  component: Slider,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
export var Default = {
  args: {
    min: 0,
    max: 100,
    step: 1,
    label: "Âm lượng"
  },
  name: "Mặc định"
};
export var ShowValue = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    showValue: true
  }),
  name: "Hiển thị giá trị"
};
export var Marks = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    max: 10,
    label: "Đánh giá",
    marks: [0, 2, 4, 6, 8, 10]
  }),
  name: "Có đánh dấu"
};
export var WithPrefixSuffix = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    step: 5,
    label: "Khoảng giá",
    prefix: "₫",
    suffix: "VNĐ"
  }),
  name: "Có tiền tố và hậu tố"
};
export var RangeSlider = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    label: "Chọn khoảng",
    defaultValue: [20, 80],
    showValue: true
  }),
  name: "Thanh trượt khoảng"
};
