// zmp-ui/esm/components/page/index.stories.js
import React from "react";
import Page from ".";
import App from "../app";
import ZMPRouter from "../router";
import Cluster from "../cluster";
import Text from "../text";
import Button from "../button";
import Icon from "../icon";
var meta = {
  title: "Layout/Page",
  component: Page,
  decorators: [function (Story) {
    return /*#__PURE__*/React.createElement(App, null, /*#__PURE__*/React.createElement(ZMPRouter, {
      memoryRouter: true
    }, /*#__PURE__*/React.createElement(Story, null)));
  }],
  tags: ["autodocs"]
};
export default meta;
export var Default = {
  name: "Mặc định",
  args: {
    children: /*#__PURE__*/React.createElement(Cluster, {
      space: "16px"
    }, /*#__PURE__*/React.createElement(Text.Title, null, "Ch\xE0o m\u1EEBng b\u1EA1n \u0111\u1EBFn v\u1EDBi Zalo Mini App"), /*#__PURE__*/React.createElement(Text, null, "Zalo Mini App l\xE0 nh\u1EEFng \u201C\u1EE9ng d\u1EE5ng nh\u1ECF\u201D ch\u1EA1y tr\u1EF1c ti\u1EBFp tr\xEAn n\u1EC1n t\u1EA3ng Zalo, t\u1EADn d\u1EE5ng ngu\u1ED3n t\xE0i nguy\xEAn c\xF3 s\u1EB5n v\u1EDBi ti\u1EC1m n\u0103ng kh\xF4ng gi\u1EDBi h\u1EA1n t\u1EEB Zalo."), /*#__PURE__*/React.createElement(Button, {
      suffixIcon: /*#__PURE__*/React.createElement(Icon, {
        icon: "zi-arrow-right"
      })
    }, "Ti\u1EBFp t\u1EE5c"))
  }
};
