// zmp-ui/cjs/components/radio/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.GroupExample = exports.Disabled = exports.Default = exports.Checked = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Form/Radio",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    label: "Radio mặc định"
  }
};
var Checked = exports.Checked = {
  name: "Chọn sẵn",
  args: {
    label: "Radio đã chọn",
    defaultChecked: true
  }
};
var Disabled = exports.Disabled = {
  name: "Trạng thái disabled",
  args: {
    label: "Radio bị vô hiệu hóa",
    disabled: true
  }
};
var GroupExample = exports.GroupExample = {
  name: "Radio Group",
  render: function render(args) {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Group, (0, _extends2["default"])({}, args, {
      defaultValue: "male",
      options: [{
        label: "Name",
        value: "male"
      }, {
        label: "Nữ",
        value: "female"
      }, {
        label: "Khác",
        value: "other"
      }]
    }));
  }
};
