// zmp-ui/cjs/components/calendar/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.StartOfWeekMonday = exports.JapaneseLocale = exports.Fullscreen = exports.DisabledWeekends = exports.Default = void 0;
var _ = _interopRequireDefault(require("."));
var meta = {
  title: "Display/Calendar",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
};
var _default = exports["default"] = meta;
var Default = exports.Default = {
  args: {},
  name: "Mặc định"
};
var Fullscreen = exports.Fullscreen = {
  args: {
    fullscreen: true
  },
  name: "Toàn màn hình"
};
var JapaneseLocale = exports.JapaneseLocale = {
  args: {
    locale: "ja"
  },
  name: "Ngôn ngữ Nhật"
};
var DisabledWeekends = exports.DisabledWeekends = {
  args: {
    disabledDate: function disabledDate(date) {
      return date.getDay() === 0 || date.getDay() === 6;
    } // Disable weekends
  },
  parameters: {
    docs: {
      source: {
        code: "<Calendar\n  disabledDate={date => date.getDay() === 0 || date.getDay() === 6}\n/>" // How to use `parameters.docs.source`: https://storybook.js.org/docs/api/doc-blocks/doc-block-source#transform
      }
    }
  },
  name: "Vô hiệu hoá cuối tuần"
};
var StartOfWeekMonday = exports.StartOfWeekMonday = {
  args: {
    startOfWeek: 1 // Start the week on Monday
  },
  name: "Tuần bắt đầu từ thứ Hai"
};
