// zmp-ui/esm/components/checkbox/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from "react";
import Checkbox from ".";
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Form/Checkbox",
  component: Checkbox,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
export var Default = {
  name: "Mặc định",
  args: {
    label: "Checkbox mặc định"
  }
};
export var Checked = {
  name: "Chọn sẵn",
  args: {
    label: "Checkbox chọn sẵn",
    defaultChecked: true
  }
};
export var Disabled = {
  name: "Bị vô hiệu hóa",
  args: {
    label: "Checkbox bị vô hiệu hóa",
    disabled: true
  }
};
export var CheckedDisabled = {
  name: "Chọn sẵn và bị vô hiệu hóa",
  args: {
    label: "Checkbox chọn sẵn và bị vô hiệu hóa",
    defaultChecked: true,
    disabled: true
  }
};
export var Indeterminate = {
  name: "Trạng thái không xác định",
  args: {
    label: "Checkbox trạng thái không xác định",
    indeterminate: true
  }
};
export var SmallSize = {
  name: "Kích thước nhỏ",
  args: {
    label: "Checkbox kích thước nhỏ",
    size: "small"
  }
};
export var WithChildren = {
  name: "Có nội dung con",
  args: {
    children: /*#__PURE__*/React.createElement("span", null, "Checkbox v\u1EDBi ", /*#__PURE__*/React.createElement("strong", null, "n\u1ED9i dung t\xF9y ch\u1EC9nh"))
  }
};
export var GroupExample = {
  name: "Ví dụ nhóm",
  render: function render() {
    return /*#__PURE__*/React.createElement(Checkbox.Group, {
      defaultValue: ["apple"],
      options: [{
        label: "Táo",
        value: "apple"
      }, {
        label: "Cam",
        value: "orange",
        disabled: true
      }, {
        label: "Chuối",
        value: "banana"
      }]
    });
  }
};
