// zmp-ui/esm/components/calendar/index.stories.js
import Calendar from ".";
var meta = {
  title: "Display/Calendar",
  component: Calendar,
  tags: ["autodocs"],
  argTypes: {}
};
export default meta;
export var Default = {
  args: {},
  name: "Mặc định"
};
export var Fullscreen = {
  args: {
    fullscreen: true
  },
  name: "Toàn màn hình"
};
export var JapaneseLocale = {
  args: {
    locale: "ja"
  },
  name: "Ngôn ngữ Nhật"
};
export var DisabledWeekends = {
  args: {
    disabledDate: function disabledDate(date) {
      return date.getDay() === 0 || date.getDay() === 6;
    } // Disable weekends
  },
  parameters: {
    docs: {
      source: {
        code: "<Calendar\n  disabledDate={date => date.getDay() === 0 || date.getDay() === 6}\n/>" // How to use `parameters.docs.source`: https://storybook.js.org/docs/api/doc-blocks/doc-block-source#transform
      }
    }
  },
  name: "Vô hiệu hoá cuối tuần"
};
export var StartOfWeekMonday = {
  args: {
    startOfWeek: 1 // Start the week on Monday
  },
  name: "Tuần bắt đầu từ thứ Hai"
};
