// zmp-ui/esm/components/radio/index.stories.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from "react";
import Radio from ".";
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Form/Radio",
  component: Radio,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
export var Default = {
  name: "Mặc định",
  args: {
    label: "Radio mặc định"
  }
};
export var Checked = {
  name: "Chọn sẵn",
  args: {
    label: "Radio đã chọn",
    defaultChecked: true
  }
};
export var Disabled = {
  name: "Trạng thái disabled",
  args: {
    label: "Radio bị vô hiệu hóa",
    disabled: true
  }
};
export var GroupExample = {
  name: "Radio Group",
  render: function render(args) {
    return /*#__PURE__*/React.createElement(Radio.Group, _extends({}, args, {
      defaultValue: "male",
      options: [{
        label: "Name",
        value: "male"
      }, {
        label: "Nữ",
        value: "female"
      }, {
        label: "Khác",
        value: "other"
      }]
    }));
  }
};
