// zmp-ui/cjs/components/box/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.NoSpace = exports.MarginAndPadding = exports.FlexBox = exports.Default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Container/Box",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Base = {
  args: {
    children: ["#dc1f18", "#34b764", "#fdd949"].map(function (color) {
      return /*#__PURE__*/_react["default"].createElement("div", {
        key: color,
        style: {
          backgroundColor: color,
          width: 40,
          height: 40
        }
      });
    }),
    style: {
      border: "1px solid #006af5"
    }
  }
};
var Default = exports.Default = {
  name: "Mặc định",
  args: _objectSpread({}, Base.args)
};
var MarginAndPadding = exports.MarginAndPadding = {
  name: "Có margin và padding",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    m: 2,
    p: 2
  })
};
var NoSpace = exports.NoSpace = {
  name: "Không có khoảng cách",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    noSpace: true
  })
};
var FlexBox = exports.FlexBox = {
  name: "Flexbox",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    flex: true,
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "row"
  })
};
