// zmp-vite-plugin/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  default: () => zaloMiniApp
});
module.exports = __toCommonJS(src_exports);
var import_fs = require("fs");
function zaloMiniApp(options) {
  const config = {
    name: "vite-plugin-zalo-mini-app",
    config(config2, env) {
      const maOptions = options ?? JSON.parse((0, import_fs.readFileSync)("./app-config.json", { encoding: "utf-8" }));
      return {
        base: env.command === "serve" ? config2.base : "./",
        build: {
          outDir: config2.build?.outDir || "www",
          target: config2.build?.target || "es2015",
          cssTarget: config2.build?.cssTarget || ["es2015", "safari13.1"],
          rollupOptions: {
            ...config2.build?.rollupOptions,
            output: {
              entryFileNames: "assets/[name].[hash].module.js",
              chunkFileNames: "assets/[name].[hash].module.js",
              ...config2.build?.rollupOptions?.output
            },
            plugins: [
              ...[config2.build?.rollupOptions?.plugins],
              {
                name: "vite-plugin-zalo-mini-app",
                writeBundle(options2, bundle) {
                  const output = Object.values(bundle);
                  const outputMap = /* @__PURE__ */ new Map();
                  output.forEach((obj) => {
                    outputMap.set(obj.fileName, obj);
                  });
                  const jsFiles = output.filter((file) => {
                    return file.type === "chunk" && file.isEntry;
                  });
                  const modulePreloadFiles = [];
                  const getImportedChunks = (chunk, seen = /* @__PURE__ */ new Set()) => {
                    const chunks = [];
                    chunk.imports.forEach((file) => {
                      const importee = outputMap.get(file);
                      if (importee?.type === "chunk" && !seen.has(file)) {
                        seen.add(file);
                        chunks.push(...getImportedChunks(importee, seen));
                        chunks.push(importee);
                      }
                    });
                    return chunks;
                  };
                  jsFiles.forEach((file) => {
                    const chunks = getImportedChunks(file);
                    modulePreloadFiles.push(...chunks);
                  });
                  const cssFiles = output.filter((file) => {
                    if (file.type !== "asset" || !file.fileName.endsWith(".css"))
                      return false;
                    return true;
                  });
                  const appConfigJson = {
                    app: maOptions.app,
                    listCSS: [
                      ...cssFiles.map((f) => f.fileName),
                      ...maOptions.listCSS || []
                    ],
                    listSyncJS: [
                      ...jsFiles.map((f) => f.fileName),
                      ...maOptions.listSyncJS || []
                    ],
                    listAsyncJS: [
                      ...modulePreloadFiles.map((f) => f.fileName),
                      ...maOptions.listAsyncJS || []
                    ],
                    pages: []
                  };
                  (0, import_fs.writeFileSync)(
                    `${options2.dir}/app-config.json`,
                    JSON.stringify(appConfigJson)
                  );
                }
              }
            ]
          }
        }
      };
    }
  };
  return config;
}
