// zmp-ui/esm/components/modal/index.stories.js
import { Avatar, Box, Text } from "../..";
import React from "react";
import Modal from ".";
var meta = {
  title: "Overlay/Modal",
  component: Modal,
  tags: ["autodocs"],
  argTypes: {}
};
export default meta;
export var InfoDialog = {
  name: "Mặc định",
  args: {
    visible: true,
    title: "Bạn có muốn đặt bàn tại ZaUI Coffee không?",
    actions: [{
      text: "Xem menu"
    }, {
      text: "Đặt bàn ngay",
      close: true,
      highLight: true
    }],
    description: "Hãy xem qua menu trước khi quyết định đặt bàn. Chúng tôi có nhiều món hấp dẫn đang chờ bạn!"
  }
};
export var InfoDialogVerticalActions = {
  name: "Hiển thị các action theo chiều dọc",
  args: {
    visible: true,
    title: "Bạn có muốn đặt bàn tại ZaUI Coffee không?",
    verticalActions: true,
    actions: [{
      text: "Xem menu chi tiết"
    }, {
      text: "Tiếp tục đặt bàn",
      close: true,
      highLight: true
    }],
    description: "Bạn có thể tham khảo thêm thông tin món ăn trước khi đặt bàn để có trải nghiệm tốt nhất."
  }
};
export var CustomModalWithCover = {
  name: "Hình ảnh cover",
  args: {
    visible: true,
    title: "Ưu đãi đặc biệt từ ZaUI Coffee",
    coverSrc: "https://images.unsplash.com/photo-1561336313-0bd5e0b27ec8",
    zIndex: 1200,
    actions: [{
      text: "Không quan tâm"
    }, {
      text: "Tìm hiểu thêm",
      close: true,
      highLight: true
    }],
    description: "Khám phá các chương trình khuyến mãi hấp dẫn dành riêng cho thành viên khi đặt bàn trực tuyến."
  }
};
export var CustomContent = {
  name: "Nội dung tùy chỉnh",
  args: {
    visible: true,
    actions: [{
      text: "Đóng",
      close: true,
      highLight: true
    }],
    coverSrc: "https://images.unsplash.com/photo-1561336313-0bd5e0b27ec8",
    children: /*#__PURE__*/React.createElement(Box, {
      flex: true,
      flexDirection: "column",
      alignItems: "center",
      textAlign: "center"
    }, /*#__PURE__*/React.createElement(Avatar, {
      style: {
        position: "absolute",
        marginTop: -60
      }
    }, "Z"), /*#__PURE__*/React.createElement(Text.Title, null, "ZaUI Coffee"), /*#__PURE__*/React.createElement(Text, {
      size: "small"
    }, "Chu\u1ED7i cafe \u0111\u01B0\u1EE3c y\xEAu th\xEDch s\u1ED1 1 t\u1EA1i Vi\u1EC7t Nam"))
  }
};
