// zmp-ui/cjs/components/checkbox/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithChildren = exports.SmallSize = exports.Indeterminate = exports.GroupExample = exports.Disabled = exports.Default = exports.CheckedDisabled = exports.Checked = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Form/Checkbox",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    label: "Checkbox mặc định"
  }
};
var Checked = exports.Checked = {
  name: "Chọn sẵn",
  args: {
    label: "Checkbox chọn sẵn",
    defaultChecked: true
  }
};
var Disabled = exports.Disabled = {
  name: "Bị vô hiệu hóa",
  args: {
    label: "Checkbox bị vô hiệu hóa",
    disabled: true
  }
};
var CheckedDisabled = exports.CheckedDisabled = {
  name: "Chọn sẵn và bị vô hiệu hóa",
  args: {
    label: "Checkbox chọn sẵn và bị vô hiệu hóa",
    defaultChecked: true,
    disabled: true
  }
};
var Indeterminate = exports.Indeterminate = {
  name: "Trạng thái không xác định",
  args: {
    label: "Checkbox trạng thái không xác định",
    indeterminate: true
  }
};
var SmallSize = exports.SmallSize = {
  name: "Kích thước nhỏ",
  args: {
    label: "Checkbox kích thước nhỏ",
    size: "small"
  }
};
var WithChildren = exports.WithChildren = {
  name: "Có nội dung con",
  args: {
    children: /*#__PURE__*/_react["default"].createElement("span", null, "Checkbox v\u1EDBi ", /*#__PURE__*/_react["default"].createElement("strong", null, "n\u1ED9i dung t\xF9y ch\u1EC9nh"))
  }
};
var GroupExample = exports.GroupExample = {
  name: "Ví dụ nhóm",
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Group, {
      defaultValue: ["apple"],
      options: [{
        label: "Táo",
        value: "apple"
      }, {
        label: "Cam",
        value: "orange",
        disabled: true
      }, {
        label: "Chuối",
        value: "banana"
      }]
    });
  }
};
