// zmp-ui/cjs/components/avatar/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithImage = exports.StorySeen = exports.StoryDefault = exports.SmallAvatar = exports.Online = exports.LargeAvatar = exports.Default = exports.Blocked = exports.BackgroundColor = exports.AvatarGroupExample = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
var meta = _objectSpread({
  title: "Display/Avatar",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineCenterParameters);
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    children: "A",
    size: 48
  }
};
var Online = exports.Online = {
  name: "Trực tuyến",
  args: {
    children: "B",
    size: 48,
    online: true
  }
};
var Blocked = exports.Blocked = {
  name: "Bị khóa",
  args: {
    children: "C",
    size: 48,
    blocked: true
  }
};
var StoryDefault = exports.StoryDefault = {
  name: "Story mới",
  args: {
    children: "D",
    size: 48,
    story: "default"
  }
};
var StorySeen = exports.StorySeen = {
  name: "Story đã xem",
  args: {
    children: "E",
    size: 48,
    story: "seen"
  }
};
var WithImage = exports.WithImage = {
  name: "Có ảnh đại diện",
  args: {
    src: "https://avatars.githubusercontent.com/u/89498710",
    size: 48
  }
};
var SmallAvatar = exports.SmallAvatar = {
  name: "Avatar nhỏ",
  args: {
    children: "F",
    size: 24
  }
};
var LargeAvatar = exports.LargeAvatar = {
  name: "Avatar lớn",
  args: {
    children: "G",
    size: 96
  }
};
var BackgroundColor = exports.BackgroundColor = {
  name: "Màu nền tùy chỉnh",
  args: {
    children: "H",
    size: 48,
    backgroundColor: "PURPLE-BLUE"
  }
};
var AvatarGroupExample = exports.AvatarGroupExample = {
  name: "Nhóm Avatar",
  render: function render(args) {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Group, (0, _extends2["default"])({}, args, {
      maxCounter: 3
    }), /*#__PURE__*/_react["default"].createElement(_["default"], null, "A"), /*#__PURE__*/_react["default"].createElement(_["default"], null, "B"), /*#__PURE__*/_react["default"].createElement(_["default"], null, "C"), /*#__PURE__*/_react["default"].createElement(_["default"], null, "D"));
  }
};
