// zmp-ui/cjs/components/grid/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.SingleColumn = exports.NoSpacing = exports.Default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Container/Grid",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Base = {
  args: {
    children: ["#dc1f18", "#34b764", "#fdd949", "#dc1f18", "#34b764", "#fdd949", "#dc1f18", "#34b764", "#fdd949", "#dc1f18", "#34b764", "#fdd949"].map(function (color) {
      return /*#__PURE__*/_react["default"].createElement("div", {
        key: color,
        style: {
          backgroundColor: color,
          width: 40,
          height: 40
        }
      });
    }),
    style: {
      width: 240,
      border: "1px solid #006af5"
    },
    columnCount: 3,
    rowSpace: "1rem",
    columnSpace: "1rem"
  }
};
var Default = exports.Default = {
  name: "Mặc định",
  args: _objectSpread({}, Base.args)
};
var SingleColumn = exports.SingleColumn = {
  name: "1 cột",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    columnCount: 1
  })
};
var NoSpacing = exports.NoSpacing = {
  name: "Không có khoảng cách",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    rowSpace: "0",
    columnSpace: "0"
  })
};
