// zmp-ui/cjs/components/progress/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithLabel = exports.Default = exports.CustomStrokeWidth = exports.CustomColors = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _ = _interopRequireDefault(require("."));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Display/Progress",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    completed: 50,
    maxCompleted: 100
  }
};
var CustomColors = exports.CustomColors = {
  name: "Màu tùy chỉnh",
  args: {
    completed: 30,
    maxCompleted: 100,
    strokeColor: "green",
    trailColor: "lightgray"
  }
};
var CustomStrokeWidth = exports.CustomStrokeWidth = {
  name: "Độ dày tùy chỉnh",
  args: {
    completed: 60,
    maxCompleted: 100,
    strokeWidth: 10
  }
};
var WithLabel = exports.WithLabel = {
  name: "Có nhãn",
  args: {
    completed: 75,
    maxCompleted: 100,
    showLabel: true
  }
};
