// zmp-ui/esm/components/switch/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import Switch from ".";
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Form/Switch",
  component: Switch,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
var label = "Nhận thông báo";
export var Default = {
  name: "Mặc định",
  args: {
    label: label
  }
};
export var Checked = {
  name: "Bật sẵn",
  args: {
    label: label,
    defaultChecked: true
  }
};
export var Disabled = {
  name: "Trạng thái disabled",
  args: {
    label: label,
    disabled: true
  }
};
export var Small = {
  name: "Size small",
  args: {
    label: label,
    size: "small"
  }
};
