// zmp-ui/cjs/components/text/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.XXXXSmallBold = exports.XXXXSmall = exports.XXXSmallBold = exports.XXXSmall = exports.XXSmallBold = exports.XXSmall = exports.XSmallBold = exports.XSmall = exports.XLargeBold = exports.XLarge = exports.TitleXLarge = exports.TitleSmall = exports.TitleLarge = exports.Title = exports.SmallBold = exports.Small = exports.LargeBold = exports.Large = exports.HeaderText = exports.HeaderSmall = exports.Default = exports.BoldText = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var meta = {
  title: "Display/Text",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {},
  parameters: {
    docs: {
      story: {
        inline: true
      }
    }
  }
};
var _default = exports["default"] = meta;
var children = "Zalo Mini App là những ứng dụng nhỏ chạy trực tiếp trên nền tảng Zalo, tận dụng nguồn tài nguyên có sẵn với tiềm năng không giới hạn từ Zalo.";
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    children: children
  }
};
var XXXXSmall = exports.XXXXSmall = {
  name: "xxxxSmall",
  args: {
    children: children,
    size: "xxxxSmall"
  }
};
var XXXSmall = exports.XXXSmall = {
  name: "xxxSmall",
  args: {
    children: children,
    size: "xxxSmall"
  }
};
var XXSmall = exports.XXSmall = {
  name: "xxSmall",
  args: {
    children: children,
    size: "xxSmall"
  }
};
var XSmall = exports.XSmall = {
  name: "xSmall",
  args: {
    children: children,
    size: "xSmall"
  }
};
var Small = exports.Small = {
  name: "small",
  args: {
    children: children,
    size: "small"
  }
};
var Large = exports.Large = {
  name: "large",
  args: {
    children: children,
    size: "large"
  }
};
var XLarge = exports.XLarge = {
  name: "xLarge",
  args: {
    children: children,
    size: "xLarge"
  }
};
var BoldText = exports.BoldText = {
  name: "In đậm",
  args: {
    children: children,
    bold: true
  }
};
var XXXXSmallBold = exports.XXXXSmallBold = {
  name: "xxxxSmall + In đậm",
  args: {
    children: children,
    size: "xxxxSmall",
    bold: true
  }
};
var XXXSmallBold = exports.XXXSmallBold = {
  name: "xxxSmall + In đậm",
  args: {
    children: children,
    size: "xxxSmall",
    bold: true
  }
};
var XXSmallBold = exports.XXSmallBold = {
  name: "xxSmall + In đậm",
  args: {
    children: children,
    size: "xxSmall",
    bold: true
  }
};
var XSmallBold = exports.XSmallBold = {
  name: "xSmall + In đậm",
  args: {
    children: children,
    size: "xSmall",
    bold: true
  }
};
var SmallBold = exports.SmallBold = {
  name: "small + In đậm",
  args: {
    children: children,
    size: "small",
    bold: true
  }
};
var LargeBold = exports.LargeBold = {
  name: "large + In đậm",
  args: {
    children: children,
    size: "large",
    bold: true
  }
};
var XLargeBold = exports.XLargeBold = {
  name: "xLarge + In đậm",
  args: {
    children: children,
    size: "xLarge",
    bold: true
  }
};
var headerChildren = "I. Zalo Mini App là gì?";
var HeaderText = exports.HeaderText = {
  name: "Chỉ mục mặc định",
  render: function render(args) {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Header, args, headerChildren);
  }
};
var HeaderSmall = exports.HeaderSmall = {
  name: "Chỉ mục small",
  render: function render(args) {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Header, (0, _extends2["default"])({}, args, {
      size: "small"
    }), "I. Zalo Mini App l\xE0 g\xEC?");
  }
};
var titleChildren = "Zalo Mini App";
var Title = exports.Title = {
  name: "Tiêu đề mặc định",
  render: function render(args) {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Title, args, titleChildren);
  }
};
var TitleSmall = exports.TitleSmall = {
  name: "Tiêu đề small",
  render: function render(args) {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Title, (0, _extends2["default"])({}, args, {
      size: "small"
    }), titleChildren);
  }
};
var TitleLarge = exports.TitleLarge = {
  name: "Tiêu đề large",
  render: function render(args) {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Title, (0, _extends2["default"])({}, args, {
      size: "large"
    }), titleChildren);
  }
};
var TitleXLarge = exports.TitleXLarge = {
  name: "Tiêu đề xLarge",
  render: function render(args) {
    return /*#__PURE__*/_react["default"].createElement(_["default"].Title, (0, _extends2["default"])({}, args, {
      size: "xLarge"
    }), titleChildren);
  }
};
