// zmp-ui/esm/components/text/index.stories.js
import _extends from "@babel/runtime/helpers/extends";
import React from "react";
import Text from ".";
var meta = {
  title: "Display/Text",
  component: Text,
  tags: ["autodocs"],
  argTypes: {},
  parameters: {
    docs: {
      story: {
        inline: true
      }
    }
  }
};
export default meta;
var children = "Zalo Mini App là những ứng dụng nhỏ chạy trực tiếp trên nền tảng Zalo, tận dụng nguồn tài nguyên có sẵn với tiềm năng không giới hạn từ Zalo.";
export var Default = {
  name: "Mặc định",
  args: {
    children: children
  }
};
export var XXXXSmall = {
  name: "xxxxSmall",
  args: {
    children: children,
    size: "xxxxSmall"
  }
};
export var XXXSmall = {
  name: "xxxSmall",
  args: {
    children: children,
    size: "xxxSmall"
  }
};
export var XXSmall = {
  name: "xxSmall",
  args: {
    children: children,
    size: "xxSmall"
  }
};
export var XSmall = {
  name: "xSmall",
  args: {
    children: children,
    size: "xSmall"
  }
};
export var Small = {
  name: "small",
  args: {
    children: children,
    size: "small"
  }
};
export var Large = {
  name: "large",
  args: {
    children: children,
    size: "large"
  }
};
export var XLarge = {
  name: "xLarge",
  args: {
    children: children,
    size: "xLarge"
  }
};
export var BoldText = {
  name: "In đậm",
  args: {
    children: children,
    bold: true
  }
};
export var XXXXSmallBold = {
  name: "xxxxSmall + In đậm",
  args: {
    children: children,
    size: "xxxxSmall",
    bold: true
  }
};
export var XXXSmallBold = {
  name: "xxxSmall + In đậm",
  args: {
    children: children,
    size: "xxxSmall",
    bold: true
  }
};
export var XXSmallBold = {
  name: "xxSmall + In đậm",
  args: {
    children: children,
    size: "xxSmall",
    bold: true
  }
};
export var XSmallBold = {
  name: "xSmall + In đậm",
  args: {
    children: children,
    size: "xSmall",
    bold: true
  }
};
export var SmallBold = {
  name: "small + In đậm",
  args: {
    children: children,
    size: "small",
    bold: true
  }
};
export var LargeBold = {
  name: "large + In đậm",
  args: {
    children: children,
    size: "large",
    bold: true
  }
};
export var XLargeBold = {
  name: "xLarge + In đậm",
  args: {
    children: children,
    size: "xLarge",
    bold: true
  }
};
var headerChildren = "I. Zalo Mini App là gì?";
export var HeaderText = {
  name: "Chỉ mục mặc định",
  render: function render(args) {
    return /*#__PURE__*/React.createElement(Text.Header, args, headerChildren);
  }
};
export var HeaderSmall = {
  name: "Chỉ mục small",
  render: function render(args) {
    return /*#__PURE__*/React.createElement(Text.Header, _extends({}, args, {
      size: "small"
    }), "I. Zalo Mini App l\xE0 g\xEC?");
  }
};
var titleChildren = "Zalo Mini App";
export var Title = {
  name: "Tiêu đề mặc định",
  render: function render(args) {
    return /*#__PURE__*/React.createElement(Text.Title, args, titleChildren);
  }
};
export var TitleSmall = {
  name: "Tiêu đề small",
  render: function render(args) {
    return /*#__PURE__*/React.createElement(Text.Title, _extends({}, args, {
      size: "small"
    }), titleChildren);
  }
};
export var TitleLarge = {
  name: "Tiêu đề large",
  render: function render(args) {
    return /*#__PURE__*/React.createElement(Text.Title, _extends({}, args, {
      size: "large"
    }), titleChildren);
  }
};
export var TitleXLarge = {
  name: "Tiêu đề xLarge",
  render: function render(args) {
    return /*#__PURE__*/React.createElement(Text.Title, _extends({}, args, {
      size: "xLarge"
    }), titleChildren);
  }
};
