// zmp-ui/esm/components/input/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from "react";
import Input from ".";
import Icon from "../icon"; // ZaUI Icon
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
import Box from "../box";
var meta = _objectSpread({
  title: "Form/Input",
  component: Input,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
var Base = {
  name: "Mặc định",
  args: {
    label: "Tên người dùng",
    helperText: "Vui lòng nhập tên người dùng của bạn."
  }
};
export var Default = {
  name: "Mặc định",
  args: _objectSpread({}, Base.args)
};
export var WithError = {
  name: "Với trạng thái lỗi",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    status: "error",
    errorText: "Tên người dùng không hợp lệ."
  })
};
export var WithClearable = {
  name: "Có thể xóa",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    clearable: true
  })
};
export var WithMaxLength = {
  name: "Với giới hạn ký tự",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    maxLength: 20,
    showCount: true
  })
};
export var Disabled = {
  name: "Bị vô hiệu hóa",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    disabled: true
  })
};
export var WithPrefixIcon = {
  name: "Với biểu tượng đầu",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    prefix: /*#__PURE__*/React.createElement(Box, {
      pl: 4
    }, /*#__PURE__*/React.createElement(Icon, {
      icon: "zi-arrow-left"
    }))
  })
};
export var WithSuffixIcon = {
  name: "Với biểu tượng cuối",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    suffix: /*#__PURE__*/React.createElement(Box, {
      pr: 4
    }, /*#__PURE__*/React.createElement(Icon, {
      icon: "zi-arrow-right"
    }))
  })
};
export var WithPrefixAndSuffixIcon = {
  name: "Với biểu tượng đầu và cuối",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    prefix: /*#__PURE__*/React.createElement(Box, {
      pl: 4
    }, /*#__PURE__*/React.createElement(Icon, {
      icon: "zi-call"
    })),
    suffix: /*#__PURE__*/React.createElement(Box, {
      pr: 4
    }, /*#__PURE__*/React.createElement(Icon, {
      icon: "zi-info-circle"
    }))
  })
};
