// zmp-ui/esm/components/input/OTP.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import OTP from "./OTP";
import { inlineCenterParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Form/OTP",
  component: OTP,
  tags: ["autodocs"],
  argTypes: {}
}, inlineCenterParameters);
export default meta;
export var Default = {
  name: "Mặc định",
  args: {
    otpLength: 4,
    defaultValue: ""
  }
};
export var WithCustomLength = {
  name: "Độ dài tùy chỉnh",
  args: {
    otpLength: 6,
    defaultValue: ""
  }
};
export var UnmaskedInput = {
  name: "Hiển thị ký tự",
  args: {
    otpLength: 4,
    defaultValue: "",
    show: true
  }
};
export var Disabled = {
  name: "Trạng thái disabled",
  args: {
    otpLength: 4,
    defaultValue: "",
    disabled: true
  }
};
export var ReadOnly = {
  name: "Chỉ đọc",
  args: {
    otpLength: 4,
    defaultValue: "1234",
    readOnly: true
  }
};
