// zmp-ui/esm/components/input/TextArea.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import TextArea from "./TextArea";
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Form/TextArea",
  component: TextArea,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
export var Default = {
  name: "Mặc định",
  args: {
    label: "Ghi chú"
  }
};
export var WithPlaceholder = {
  name: "Với placeholder",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    placeholder: "Giao hàng trước 5h chiều"
  })
};
export var DefaultValue = {
  name: "Giá trị mặc định",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    defaultValue: "Nội dung cũ"
  })
};
export var AutoHeight = {
  name: "Tự động điều chỉnh chiều cao",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    autoHeight: true
  })
};
export var WithCharacterCount = {
  name: "Với bộ đếm ký tự",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    maxLength: 100,
    showCount: true
  })
};
export var WithHelperText = {
  name: "Với văn bản trợ giúp",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    helperText: "Vui lòng nhập không quá 100 ký tự."
  })
};
export var WithErrorStatus = {
  name: "Với trạng thái lỗi",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    status: "error",
    errorText: "Đã xảy ra lỗi. Vui lòng kiểm tra lại nội dung."
  })
};
export var Disabled = {
  name: "Bị vô hiệu hóa",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    disabled: true
  })
};
