// zmp-ui/cjs/components/input/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithSuffixIcon = exports.WithPrefixIcon = exports.WithPrefixAndSuffixIcon = exports.WithMaxLength = exports.WithError = exports.WithClearable = exports.Disabled = exports.Default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _icon = _interopRequireDefault(require("../icon"));
var _parameters = require("../../../../.storybook/parameters");
var _box = _interopRequireDefault(require("../box"));
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; } // ZaUI Icon
var meta = _objectSpread({
  title: "Form/Input",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Base = {
  name: "Mặc định",
  args: {
    label: "Tên người dùng",
    helperText: "Vui lòng nhập tên người dùng của bạn."
  }
};
var Default = exports.Default = {
  name: "Mặc định",
  args: _objectSpread({}, Base.args)
};
var WithError = exports.WithError = {
  name: "Với trạng thái lỗi",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    status: "error",
    errorText: "Tên người dùng không hợp lệ."
  })
};
var WithClearable = exports.WithClearable = {
  name: "Có thể xóa",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    clearable: true
  })
};
var WithMaxLength = exports.WithMaxLength = {
  name: "Với giới hạn ký tự",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    maxLength: 20,
    showCount: true
  })
};
var Disabled = exports.Disabled = {
  name: "Bị vô hiệu hóa",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    disabled: true
  })
};
var WithPrefixIcon = exports.WithPrefixIcon = {
  name: "Với biểu tượng đầu",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    prefix: /*#__PURE__*/_react["default"].createElement(_box["default"], {
      pl: 4
    }, /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-arrow-left"
    }))
  })
};
var WithSuffixIcon = exports.WithSuffixIcon = {
  name: "Với biểu tượng cuối",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    suffix: /*#__PURE__*/_react["default"].createElement(_box["default"], {
      pr: 4
    }, /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-arrow-right"
    }))
  })
};
var WithPrefixAndSuffixIcon = exports.WithPrefixAndSuffixIcon = {
  name: "Với biểu tượng đầu và cuối",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    prefix: /*#__PURE__*/_react["default"].createElement(_box["default"], {
      pl: 4
    }, /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-call"
    })),
    suffix: /*#__PURE__*/_react["default"].createElement(_box["default"], {
      pr: 4
    }, /*#__PURE__*/_react["default"].createElement(_icon["default"], {
      icon: "zi-info-circle"
    }))
  })
};
