// zmp-ui/cjs/components/input/OTP.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithCustomLength = exports.UnmaskedInput = exports.ReadOnly = exports.Disabled = exports.Default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _OTP = _interopRequireDefault(require("./OTP"));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Form/OTP",
  component: _OTP["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineCenterParameters);
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    otpLength: 4,
    defaultValue: ""
  }
};
var WithCustomLength = exports.WithCustomLength = {
  name: "Độ dài tùy chỉnh",
  args: {
    otpLength: 6,
    defaultValue: ""
  }
};
var UnmaskedInput = exports.UnmaskedInput = {
  name: "Hiển thị ký tự",
  args: {
    otpLength: 4,
    defaultValue: "",
    show: true
  }
};
var Disabled = exports.Disabled = {
  name: "Trạng thái disabled",
  args: {
    otpLength: 4,
    defaultValue: "",
    disabled: true
  }
};
var ReadOnly = exports.ReadOnly = {
  name: "Chỉ đọc",
  args: {
    otpLength: 4,
    defaultValue: "1234",
    readOnly: true
  }
};
