// zmp-ui/cjs/components/input/TextArea.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithPlaceholder = exports.WithHelperText = exports.WithErrorStatus = exports.WithCharacterCount = exports.Disabled = exports.DefaultValue = exports.Default = exports.AutoHeight = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _TextArea = _interopRequireDefault(require("./TextArea"));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Form/TextArea",
  component: _TextArea["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    label: "Ghi chú"
  }
};
var WithPlaceholder = exports.WithPlaceholder = {
  name: "Với placeholder",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    placeholder: "Giao hàng trước 5h chiều"
  })
};
var DefaultValue = exports.DefaultValue = {
  name: "Giá trị mặc định",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    defaultValue: "Nội dung cũ"
  })
};
var AutoHeight = exports.AutoHeight = {
  name: "Tự động điều chỉnh chiều cao",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    autoHeight: true
  })
};
var WithCharacterCount = exports.WithCharacterCount = {
  name: "Với bộ đếm ký tự",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    maxLength: 100,
    showCount: true
  })
};
var WithHelperText = exports.WithHelperText = {
  name: "Với văn bản trợ giúp",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    helperText: "Vui lòng nhập không quá 100 ký tự."
  })
};
var WithErrorStatus = exports.WithErrorStatus = {
  name: "Với trạng thái lỗi",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    status: "error",
    errorText: "Đã xảy ra lỗi. Vui lòng kiểm tra lại nội dung."
  })
};
var Disabled = exports.Disabled = {
  name: "Bị vô hiệu hóa",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    disabled: true
  })
};
