// zmp-ui/cjs/components/swiper/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithLoopAndAutoplay = exports.Default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = {
  title: "Display/Swiper",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
};
var _default = exports["default"] = meta;
function SlideContent() {
  return /*#__PURE__*/_react["default"].createElement("img", {
    src: "https://stc-zmp.zadn.vn/zmp-zaui/images/0e05d63a7a93a6cdff826.jpg",
    alt: "Swiper Slide",
    style: {
      width: "100%"
    }
  });
}
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    children: [/*#__PURE__*/_react["default"].createElement(_["default"].Slide, null, /*#__PURE__*/_react["default"].createElement(SlideContent, null)), /*#__PURE__*/_react["default"].createElement(_["default"].Slide, null, /*#__PURE__*/_react["default"].createElement(SlideContent, null)), /*#__PURE__*/_react["default"].createElement(_["default"].Slide, null, /*#__PURE__*/_react["default"].createElement(SlideContent, null))]
  }
};
var WithLoopAndAutoplay = exports.WithLoopAndAutoplay = {
  name: "Autoplay và loop",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    autoplay: true,
    loop: true
  })
};
