// zmp-ui/esm/components/spinner/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import Spinner from ".";
import { inlineCenterParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Display/Spinner",
  component: Spinner,
  tags: ["autodocs"],
  argTypes: {}
}, inlineCenterParameters);
export default meta;
export var Default = {
  name: "Mặc định",
  args: {}
};
export var Invisible = {
  name: "Không hiển thị",
  args: {
    visible: false
  },
  parameters: {
    docs: {
      source: {
        code: "<Spinner visible={false} />"
      }
    }
  }
};
export var WithLogo = {
  args: {
    logo: "https://photo-logo-mapps.zadn.vn/2d2a6800cc45251b7c54.jpg"
  }
};
