// zmp-ui/cjs/components/page/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _app = _interopRequireDefault(require("../app"));
var _router = _interopRequireDefault(require("../router"));
var _cluster = _interopRequireDefault(require("../cluster"));
var _text = _interopRequireDefault(require("../text"));
var _button = _interopRequireDefault(require("../button"));
var _icon = _interopRequireDefault(require("../icon"));
var meta = {
  title: "Layout/Page",
  component: _["default"],
  decorators: [function (Story) {
    return /*#__PURE__*/_react["default"].createElement(_app["default"], null, /*#__PURE__*/_react["default"].createElement(_router["default"], {
      memoryRouter: true
    }, /*#__PURE__*/_react["default"].createElement(Story, null)));
  }],
  tags: ["autodocs"]
};
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    children: /*#__PURE__*/_react["default"].createElement(_cluster["default"], {
      space: "16px"
    }, /*#__PURE__*/_react["default"].createElement(_text["default"].Title, null, "Ch\xE0o m\u1EEBng b\u1EA1n \u0111\u1EBFn v\u1EDBi Zalo Mini App"), /*#__PURE__*/_react["default"].createElement(_text["default"], null, "Zalo Mini App l\xE0 nh\u1EEFng \u201C\u1EE9ng d\u1EE5ng nh\u1ECF\u201D ch\u1EA1y tr\u1EF1c ti\u1EBFp tr\xEAn n\u1EC1n t\u1EA3ng Zalo, t\u1EADn d\u1EE5ng ngu\u1ED3n t\xE0i nguy\xEAn c\xF3 s\u1EB5n v\u1EDBi ti\u1EC1m n\u0103ng kh\xF4ng gi\u1EDBi h\u1EA1n t\u1EEB Zalo."), /*#__PURE__*/_react["default"].createElement(_button["default"], {
      suffixIcon: /*#__PURE__*/_react["default"].createElement(_icon["default"], {
        icon: "zi-arrow-right"
      })
    }, "Ti\u1EBFp t\u1EE5c"))
  }
};
