// zmp-ui/esm/components/avatar/index.stories.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from "react";
import Avatar from ".";
import { inlineCenterParameters } from "../../../../.storybook/parameters";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
var meta = _objectSpread({
  title: "Display/Avatar",
  component: Avatar,
  tags: ["autodocs"],
  argTypes: {}
}, inlineCenterParameters);
export default meta;
export var Default = {
  name: "Mặc định",
  args: {
    children: "A",
    size: 48
  }
};
export var Online = {
  name: "Trực tuyến",
  args: {
    children: "B",
    size: 48,
    online: true
  }
};
export var Blocked = {
  name: "Bị khóa",
  args: {
    children: "C",
    size: 48,
    blocked: true
  }
};
export var StoryDefault = {
  name: "Story mới",
  args: {
    children: "D",
    size: 48,
    story: "default"
  }
};
export var StorySeen = {
  name: "Story đã xem",
  args: {
    children: "E",
    size: 48,
    story: "seen"
  }
};
export var WithImage = {
  name: "Có ảnh đại diện",
  args: {
    src: "https://avatars.githubusercontent.com/u/89498710",
    size: 48
  }
};
export var SmallAvatar = {
  name: "Avatar nhỏ",
  args: {
    children: "F",
    size: 24
  }
};
export var LargeAvatar = {
  name: "Avatar lớn",
  args: {
    children: "G",
    size: 96
  }
};
export var BackgroundColor = {
  name: "Màu nền tùy chỉnh",
  args: {
    children: "H",
    size: 48,
    backgroundColor: "PURPLE-BLUE"
  }
};
export var AvatarGroupExample = {
  name: "Nhóm Avatar",
  render: function render(args) {
    return /*#__PURE__*/React.createElement(Avatar.Group, _extends({}, args, {
      maxCounter: 3
    }), /*#__PURE__*/React.createElement(Avatar, null, "A"), /*#__PURE__*/React.createElement(Avatar, null, "B"), /*#__PURE__*/React.createElement(Avatar, null, "C"), /*#__PURE__*/React.createElement(Avatar, null, "D"));
  }
};
