// zmp-ui/esm/components/grid/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from "react";
import Grid from ".";
import { inlineFullwidthParameters } from "../../../../.storybook/parameters";
var meta = _objectSpread({
  title: "Container/Grid",
  component: Grid,
  tags: ["autodocs"],
  argTypes: {}
}, inlineFullwidthParameters);
export default meta;
var Base = {
  args: {
    children: ["#dc1f18", "#34b764", "#fdd949", "#dc1f18", "#34b764", "#fdd949", "#dc1f18", "#34b764", "#fdd949", "#dc1f18", "#34b764", "#fdd949"].map(function (color) {
      return /*#__PURE__*/React.createElement("div", {
        key: color,
        style: {
          backgroundColor: color,
          width: 40,
          height: 40
        }
      });
    }),
    style: {
      width: 240,
      border: "1px solid #006af5"
    },
    columnCount: 3,
    rowSpace: "1rem",
    columnSpace: "1rem"
  }
};
export var Default = {
  name: "Mặc định",
  args: _objectSpread({}, Base.args)
};
export var SingleColumn = {
  name: "1 cột",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    columnCount: 1
  })
};
export var NoSpacing = {
  name: "Không có khoảng cách",
  args: _objectSpread(_objectSpread({}, Base.args), {}, {
    rowSpace: "0",
    columnSpace: "0"
  })
};
