// zmp-ui/cjs/components/header/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.Default = exports.CustomColor = void 0;
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("."));
var _router = _interopRequireDefault(require("../router"));
var _page = _interopRequireDefault(require("../page"));
var meta = {
  title: "Layout/Header",
  component: _["default"],
  decorators: function decorators(Story) {
    return /*#__PURE__*/_react["default"].createElement(_router["default"], {
      memoryRouter: true
    }, /*#__PURE__*/_react["default"].createElement(_page["default"], null, /*#__PURE__*/_react["default"].createElement(Story, null)));
  },
  tags: ["autodocs"],
  argTypes: {
    showBackIcon: {
      control: "boolean",
      defaultValue: true,
      type: "boolean"
    }
  }
};
var _default = exports["default"] = meta;
var Default = exports.Default = {
  name: "Mặc định",
  args: {
    title: "Trang chủ"
  }
};
var CustomColor = exports.CustomColor = {
  name: "Tuỳ chỉnh màu",
  args: {
    title: "Trang chủ",
    backgroundColor: "#52b361",
    textColor: "#ffffff"
  }
};
