// zmp-ui/esm/components/tabs/index.stories.js
import React from "react";
import Tabs from ".";
import Box from "../box";
import Text from "../text";
var meta = {
  title: "Layout/Tabs",
  component: Tabs,
  tags: ["autodocs"],
  argTypes: {}
};
export default meta;
var decorators = function decorators(Story) {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      position: "fixed",
      inset: 0
    }
  }, /*#__PURE__*/React.createElement(Story, null));
};
export var Default = {
  name: "Mặc định",
  decorators: decorators,
  args: {
    children: [/*#__PURE__*/React.createElement(Tabs.Tab, {
      key: "tab1",
      label: "Gi\u1EDBi thi\u1EC7u"
    }, /*#__PURE__*/React.createElement(Box, {
      p: 4
    }, /*#__PURE__*/React.createElement(Text, null, "Zalo Mini App l\xE0 nh\u1EEFng \u201C\u1EE9ng d\u1EE5ng nh\u1ECF\u201D ch\u1EA1y tr\u1EF1c ti\u1EBFp tr\xEAn n\u1EC1n t\u1EA3ng Zalo, t\u1EADn d\u1EE5ng ngu\u1ED3n t\xE0i nguy\xEAn c\xF3 s\u1EB5n v\u1EDBi ti\u1EC1m n\u0103ng kh\xF4ng gi\u1EDBi h\u1EA1n t\u1EEB Zalo."))), /*#__PURE__*/React.createElement(Tabs.Tab, {
      key: "tab2",
      label: "Tri\u1EC3n khai"
    }, /*#__PURE__*/React.createElement(Box, {
      p: 4
    }, /*#__PURE__*/React.createElement(Text, null, "\u0110\xE2y l\xE0 b\u01B0\u1EDBc \u0111\u1EA7u ti\xEAn \u0111\u1EC3 tri\u1EC3n khai m\u1ED9t Zalo Mini App"))), /*#__PURE__*/React.createElement(Tabs.Tab, {
      key: "tab3",
      label: "Ph\xE1t h\xE0nh"
    }, /*#__PURE__*/React.createElement(Box, {
      p: 4
    }, /*#__PURE__*/React.createElement(Text, null, "\u0110\u1EC3 ph\xE1t h\xE0nh \u1EE9ng d\u1EE5ng Zalo Mini App c\u1EE7a b\u1EA1n, vui l\xF2ng truy c\u1EADp Trang Qu\u1EA3n L\xFD Zalo Mini App, ch\u1ECDn Danh s\xE1ch phi\xEAn b\u1EA3n trong m\u1EE5c Qu\u1EA3n l\xFD phi\xEAn b\u1EA3n.")))]
  }
};
