// zmp-ui/cjs/components/slider/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = exports.WithPrefixSuffix = exports.ShowValue = exports.RangeSlider = exports.Marks = exports.Default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _ = _interopRequireDefault(require("."));
var _parameters = require("../../../../.storybook/parameters");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2["default"])(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var meta = _objectSpread({
  title: "Form/Slider",
  component: _["default"],
  tags: ["autodocs"],
  argTypes: {}
}, _parameters.inlineFullwidthParameters);
var _default = exports["default"] = meta;
var Default = exports.Default = {
  args: {
    min: 0,
    max: 100,
    step: 1,
    label: "Âm lượng"
  },
  name: "Mặc định"
};
var ShowValue = exports.ShowValue = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    showValue: true
  }),
  name: "Hiển thị giá trị"
};
var Marks = exports.Marks = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    max: 10,
    label: "Đánh giá",
    marks: [0, 2, 4, 6, 8, 10]
  }),
  name: "Có đánh dấu"
};
var WithPrefixSuffix = exports.WithPrefixSuffix = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    step: 5,
    label: "Khoảng giá",
    prefix: "₫",
    suffix: "VNĐ"
  }),
  name: "Có tiền tố và hậu tố"
};
var RangeSlider = exports.RangeSlider = {
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    label: "Chọn khoảng",
    defaultValue: [20, 80],
    showValue: true
  }),
  name: "Thanh trượt khoảng"
};
