// zmp-ui/esm/components/swiper/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from "react";
import Swiper from ".";
var meta = {
  title: "Display/Swiper",
  component: Swiper,
  tags: ["autodocs"],
  argTypes: {}
};
export default meta;
function SlideContent() {
  return /*#__PURE__*/React.createElement("img", {
    src: "https://stc-zmp.zadn.vn/zmp-zaui/images/0e05d63a7a93a6cdff826.jpg",
    alt: "Swiper Slide",
    style: {
      width: "100%"
    }
  });
}
export var Default = {
  name: "Mặc định",
  args: {
    children: [/*#__PURE__*/React.createElement(Swiper.Slide, null, /*#__PURE__*/React.createElement(SlideContent, null)), /*#__PURE__*/React.createElement(Swiper.Slide, null, /*#__PURE__*/React.createElement(SlideContent, null)), /*#__PURE__*/React.createElement(Swiper.Slide, null, /*#__PURE__*/React.createElement(SlideContent, null))]
  }
};
export var WithLoopAndAutoplay = {
  name: "Autoplay và loop",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    autoplay: true,
    loop: true
  })
};
