// zmp-ui/esm/components/picker/index.stories.js
import Picker from ".";
var meta = {
  title: "Form/Picker",
  component: Picker,
  parameters: {
    layout: "centered",
    docs: {
      story: {
        inline: false
      }
    }
  },
  tags: ["autodocs"],
  argTypes: {}
};
export default meta;
export var SingleColumn = {
  name: "Một cột",
  args: {
    label: "Địa chỉ thường trú",
    helperText: "Chọn thành phố nơi bạn đang sống",
    mask: true,
    maskClosable: true,
    title: "Chọn thành phố",
    action: {
      text: "Đóng",
      close: true
    },
    data: [{
      options: [{
        value: 1,
        displayName: "Hà Nội"
      }, {
        value: 2,
        displayName: "Hồ Chí Minh"
      }, {
        value: 3,
        displayName: "Đà Nẵng"
      }, {
        value: 4,
        displayName: "Cần Thơ"
      }, {
        value: 5,
        displayName: "Hải Phòng"
      }],
      name: "city"
    }]
  }
};
export var MultipleColumns = {
  name: "Nhiều cột",
  args: {
    label: "Chọn sản phẩm",
    helperText: "Vui lòng chọn danh mục, thương hiệu và sản phẩm",
    mask: true,
    maskClosable: true,
    title: "Bộ lọc sản phẩm",
    action: {
      text: "Xác nhận",
      close: true
    },
    data: [{
      options: [{
        value: 1,
        displayName: "Điện thoại"
      }, {
        value: 2,
        displayName: "Laptop"
      }, {
        value: 3,
        displayName: "Máy tính bảng"
      }],
      name: "category"
    }, {
      options: [{
        value: 1,
        displayName: "Apple"
      }, {
        value: 2,
        displayName: "Samsung"
      }, {
        value: 3,
        displayName: "Dell"
      }],
      name: "brand"
    }, {
      options: [{
        value: 1,
        displayName: "iPhone 14"
      }, {
        value: 2,
        displayName: "Galaxy S23"
      }, {
        value: 3,
        displayName: "XPS 13"
      }],
      name: "product"
    }]
  }
};
