// zmp-ui/esm/components/image-viewer/index.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import ImageViewer from ".";
var meta = {
  title: "Display/ImageViewer",
  component: ImageViewer,
  tags: ["autodocs"],
  argTypes: {}
};
export default meta;
var image = {
  src: "https://stc-zmp.zadn.vn/zmp-zaui/images/0e05d63a7a93a6cdff826.jpg",
  alt: "Example Image"
};
export var Default = {
  name: "Mặc định",
  args: {
    images: [image, image, image],
    visible: true
  }
};
export var CustomMaskStyle = {
  name: "Tuỳ chỉnh mask",
  args: _objectSpread(_objectSpread({}, Default.args), {}, {
    maskStyle: {
      backgroundImage: "url('https://stc-zmp.zadn.vn/zmp-zaui/images/0e05d63a7a93a6cdff826.jpg')",
      filter: "blur(10px)"
    }
  })
};
